inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -210,25 +210,27 @@ def physics_data_an_is_primary_category(record):
     return False
 
 
-def _is_core(record, only_secondary_categories=False):
-    if only_secondary_categories:
-        record_core_categories = LiteratureReader(record).arxiv_categories[1:]
-    else:
-        record_core_categories = LiteratureReader(record).arxiv_categories
+def _is_first_category_core(record):
+    record_core_categories = LiteratureReader(record).arxiv_categories[0]
+    return set([record_core_categories]) & \
+        set(current_app.config.get('ARXIV_CATEGORIES', {}).get('core'))
```

Comment:
An opportunity to clean up a bit the code :)
```suggestion
    record_core_category = LiteratureReader(record).arxiv_categories[0]
    arxiv_core_categories = set(current_app.config.get('ARXIV_CATEGORIES', {}).get('core', []))
    return record_core_category in arxiv_core_categories
```